ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md

```diff
@@ -204,10 +205,92 @@ You might be thinking, Why use a scope when you can write a class method to do t
   ...
 ```
 
-See the Additional Resources section for links to some posts that dig a bit deeper into the use cases for these two.
+See the [Additional Resources section](#additional-resources) for links to some posts that dig a bit deeper into the use cases for these two.
 
 How much do you need to understand or care about scopes? In the early going, you probably won't run into them or see why to use them. Keep them in the back of your mind for when you start working on some slightly more complicated projects that might need them.
 
+### Enums
+
+Enums (short for "enumerations") map symbolic names to integer constants. They
+make code more readable and maintainable, and they offer a performance boost
+since queries involving integers are faster than those involving strings.
+
+Enums are perfect for representing the state of an attribute that has a [discrete
+value](https://en.wikipedia.org/wiki/Continuous_or_discrete_variable). As an
+example, consider a light switch: it is either in an on or an off state. It is
+never between these two states, and it is never both on and off.
+
+#### How to use enums
+
+To implement `enum`s, we need to declare them in the model and add a column to
+store them in the database table that stores instances of the model.
+
+Enums are declared in the model's class as either an array or a hash. The hash flavor has a slight advantage
+in that the integer values mapped to the `enum` are independent of their
+position and the values stored in the database are explicitly stated.
+
+```ruby
+# Array declaration
+class LightSwitch < ApplicationRecord
+  belongs_to :room
+  enum :status, [:off, :on]           # Value depends on index of symbol
+end
+
+# Hash declaration (recommended)
+class LightSwitch < ApplicationRecord
+  belongs_to :room
+  enum :status, { off: 0, on: 1 }     # Value independent of position; explicit
+end
+```
+
+These are functionally equivalent to the following [scopes](#scopes):
+
+```ruby
+scope :on, -> { where(status: "on") }
+scope :off, -> { where(status: "off") }
+```
+
+And, just like scopes, this exposes a number of class methods that return
+collections:
+
+```ruby
```

Comment:
Nit: I think it would be worth switching this example and the scopes example above this one around. First demonstrating the class methods and then tying them back to scopes seems like the better flow.